### PR TITLE
Swagger For new rdbms API change for new biz model

### PR DESCRIPTION
**Pull request by QingqingYuan**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] This is a redo of previously signed-off pull request https://github.com/Azure/azure-rest-api-specs/pull/2207. The main change is we will still release the API as preview. Structure the folder following the guidance @ https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring#stable-and-preview. 
Also, we took out the VirturalNetworkRules which still in private preview. 
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 1]
```diff
@@ -0,0 +1,1661 @@
...
+        },
+        "sslEnforcement": {
+          "$ref": "#/definitions/SslEnforcement",
+          "description": "Enable ssl enforcement or not when connect to server."
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Storage profile of a server."
+        },
+        "sku": {
```
Comment (Suna-MS) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Remove sku from properties, please understand the request structure firstly.
{  
   "Location":"centralus",
   "Properties":{  
     "storageProfile": {
       ...
     },
     ...
   },
   "Sku":{
     ...
   },
   "Tags":{  
      ...
   }
}

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 2]
```diff
@@ -0,0 +1,1661 @@
...
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU (pricing tier) of the server."
+        },
+        "tags": {
```
Comment (Suna-MS) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Same as above, remove tags from properties.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 3]
```diff
@@ -0,0 +1,1662 @@
...
+          "x-ms-enum": {
+            "name": "ServerState",
+            "modelAsString": true
+          }
+        },
+        "fullyQualifiedDomainName": {
+          "type": "string",
+          "description": "The fully qualified domain name of a server."
+        },
+        "EarliestRestoreDate": {
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Lowercase for the first character for earliestRestoreDate

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 4]
```diff
@@ -0,0 +1,1662 @@
...
+        },
+        "sslEnforcement": {
+          "$ref": "#/definitions/SslEnforcement",
+          "description": "Enable ssl enforcement or not when connect to server."
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Storage profile of a server."
+        },
+        "sku": {
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: remove sku

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 5]
```diff
@@ -0,0 +1,1662 @@
...
+        },
+        "edition": {
+          "type": "string",
+          "description": "Edition of the performance tier."
+        },
+        "dtu": {
+          "type": "integer",
+          "description": "Database throughput unit associated with the service level objective"
+        },
+        "storageMB": {
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Still use dtu/storageMB in new biz model?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 6]
```diff
@@ -0,0 +1,1662 @@
...
+      },
+      "description": "Service level objectives for performance tier."
+    },
+    "PerformanceTierProperties": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ID of the performance tier."
+        },
+        "backupRetentionDays": {
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: min/maxBackupRetentionDays are used in examples, please double check the definitions.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 1 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Thanks for catching this! I was only took from Reva's pull request and removed the VNET part. Seems this was missed from his. Removed this.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 2 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Removed

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 3 above)
Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Updated

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 4 above)
Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Removed

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 7]
```diff
@@ -0,0 +1,1662 @@
...
+        },
+        "storageProfile": {
+          "$ref": "#/definitions/StorageProfile",
+          "description": "Storage profile of a server."
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU (pricing tier) of the server."
+        },
+        "tags": {
```
[the remark this answers is not in the document]
Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Removed

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 5 above)
Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: missed in this one as well. Updated.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 6 above)
Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Reva missed this. Added.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 8]
```diff
@@ -0,0 +1,1725 @@
...
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}": {
+      "put": {
+        "tags": [
+          "Servers"
+        ],
+        "operationId": "Servers_CreateOrUpdate",
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: For PG, the operationId is Servers_Create, please make them the same for mysql and pg.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 9]
```diff
@@ -0,0 +1,1725 @@
...
+    },
+    "StorageProfile": {
+      "properties": {
+        "backupRetentionDays": {
+          "type": "integer",
+          "description": "Backup retention days for the server."
+        },
+        "geoRedundantBackup": {
+          "type": "string",
+          "description": "The tier of the particular SKU, e.g. Basic.",
```
Comment (Suna-MS) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Please update the description

Comment (QingqingYuan) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: That's already done in my local branch. i'll push it.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 10]
```diff
@@ -0,0 +1,1725 @@
...
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBforPostgreSQL/servers": {
+      "get": {
+        "tags": [
+          "Servers"
+        ],
+        "operationId": "Servers_List",
+        "x-ms-examples": {
+          "ServerList": {
+            "$ref": "./examples/ServerList.json"
```
Comment (marstr) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: As mentioned offline, there are casing issues here that are polluting CI. Let's get those fixed up, and I'll review again.

**Review by marstr — CHANGES_REQUESTED** [review]

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 10 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Yes, Thanks Martin! After update the folder name, it's passing!

**Review by marstr — CHANGES_REQUESTED** [review]
Just took a look at some of the allowed-failures in CI. It looks like some of the examples contain fields that aren't documented in the Swaggers. Mind taking a look at that?
 https://travis-ci.org/Azure/azure-rest-api-specs/jobs/338748743#L797

**Comment by QingqingYuan** [comment]
@marstr I've corrected all errors in the job you mentioned. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/338866706.

**Review by marstr — COMMENTED** [review]
I have validated that CI is passing, and that parameters other than "API Version" and "SubscriptionID" have the parameter-location specified:
https://github.com/Azure/azure-rest-api-specs/pull/2414/files#diff-6989ec44241073e88787db3d8455e5d2R1661

Because these are all additive changes, there is no concern of breaking changes.

For those reasons, these changes look good from my perspective. However, we'll still need someone from the ARM team to give their approval.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 11]
```diff
@@ -0,0 +1,1725 @@
...
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "The list of resource provider operations."
+        }
+      },
+      "description": "A list of resource provider operations."
+    },
+    "LogFileProperties": {
+      "properties": {
+        "name": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: name is repeated so many times. Its in proxy resource model, then another top level property called name, and then again here. Please keep just the one under the Proxy rsource reference.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 12]
```diff
@@ -0,0 +1,1725 @@
...
+          "type": "string",
+          "description": "Log file name."
+        },
+        "sizeInKB": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Size of the log file."
+        },
+        "createdTime": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: read only

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 13]
```diff
@@ -0,0 +1,1725 @@
...
+          "type": "string",
+          "format": "date-time",
+          "description": "Creation timestamp of the log file."
+        },
+        "lastModifiedTime": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: read only

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 14]
```diff
@@ -0,0 +1,1725 @@
...
+          "type": "string",
+          "format": "date-time",
+          "description": "Last modified timestamp of the log file."
+        },
+        "type": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: please rename this to fileType. What are the allowed values here?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 15]
```diff
@@ -0,0 +1,1725 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the configuration."
+        },
+        "defaultValue": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Default value of the configuration."
+        },
+        "dataType": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: what are the possible values? should this be an enum?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 16]
```diff
@@ -0,0 +1,1725 @@
...
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DatabaseProperties",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a database."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Is database really a proxy resource? Databases under Microsoft.SQL is a tracked resource. So it has location, tags properties also.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 17]
```diff
@@ -0,0 +1,1725 @@
...
+            "$ref": "#/definitions/FirewallRule"
+          },
+          "description": "The list of firewall rules in a server."
+        }
+      },
+      "description": "A list of firewall rules."
+    },
+    "DatabaseProperties": {
+      "properties": {
+        "charset": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Isnt this an enum?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 18]
```diff
@@ -0,0 +1,1725 @@
...
+          "type": "string",
+          "description": "The charset of the database."
+        },
+        "collation": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Does this have a fixed set of valueS?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json [hunk 19]
```diff
@@ -0,0 +1,1725 @@
...
+              "$ref": "#/definitions/ConfigurationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/logFiles": {
```
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: This actually provides a pointer to the logFile. Should this be available to anyone who has read access on the server? Or should this be exposed via a POST action

**Comment by ravbhatnagar** [comment]
@johanste  @NiklasGustafsson  - have folks from azure api review board seen this? Please review if not already.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 16 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: rdbms only has servers as tracked resource. It only has "database" as conceptually and we only having them as proxy resource

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 17 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: synced offline. We didn't change this. it's a string.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 18 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: No, not presently.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 15 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: synced offline. We didn't change this. This was in previous version. Need backend support. will not update in this iteration.

**Comment by marstr** [comment]
> @ravbhatnagar says:
@johanste @NiklasGustafsson - have folks from azure api review board seen this? Please review if not already.

I reached out to @johanste offline to get advice on how to proceed here.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 11 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Will update the API and then swagger post GA.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 13 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: fixed

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 12 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: updated

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 14 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: Will update API post GA

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json (hunk 19 above)
Comment (QingqingYuan) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2017-12-01-preview/postgresql.json]: yes, right now, there's no limitation on accessing this once the user has permission to subscription. Will consider adding more granularly control post GA.

**Comment by ravbhatnagar** [comment]
Looks good.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
Once the 20th rolls around, I'll remove the DoNotMerge label, and ping @QingqingYuan about whether or not the service team is ready for this to be merged.

**Comment by marstr** [comment]
I got word from @QingqingYuan that this has been delayed until the 28th.

**Comment by marstr** [comment]
I just got word from @QingqingYuan that the service is now available.

**Pull request merged**